packages/driver/src/cy/commands/xhr.js

```diff
@@ -471,6 +471,13 @@ module.exports = (Commands, Cypress, cy, state, config) => {
           })
         }
 
+        // look ahead to see if fixture exists
+        const fixturesRe = /^(fx:|fixture:)/
+
+        if (hasResponse && fixturesRe.test(options.response)) {
+          return cy.fixture(options.response.replace(fixturesRe, '')).then(() => route())
```

Comment:
question here: where is the loaded fixture going? I see you load it using `cy.fixture()` but shouldn't you feed it into `route` command?

Reply:
My first though was that, this fragment should only try to load a fixture and throw error if necessary. The second though was that, after sucessfull load the fixture data could be assigned as options.response. I am not sure, but with the second option the code on server side (mocking requests xhr) could be deleted, because all fixtures will be loaded here (in route command).

I wasn't sure if this approach is ok with the big picture of cypress architecture.

Please correct me/guide me if I am wrong :)

Reply:
@bahmutov I've checked if I could use loaded fixture data. I can, but only for "truthy" values. cypress server will not stub a request when `option.response` is falsy.

Moreover, I've checked the case with `false.json` file and it is failing because of the:

```
        // when data is null, JSON.stringify returns null.
        // handle that case.
        if (data === null) {
          data = ''
        }

        const chunk = Buffer.from(data, encoding)
```

I am just wondering, why cypress returns an empty string if someone wanted to stub the network request and expects the `null` value. If it is for a reason it should be also extended with `data === false` check < without it, users cannot stub a network request with `false` as a content of json file.

I think we could leave the current solution as it is, and try to address `false/null` problems in separate ticket/MR.

What do you think?